### DISCUSSION TITLE
Support Niri

### DISCUSSION THREAD

**Discussion by VarLad** [ldelossa/way-shell#35] [Announcements]
Is it possible to use way-shell out of the box with https://github.com/YaLTeR/niri/ ? What would be needed?

**Comment by ldelossa** [comment]
Not too sure. 

Do you want to try to run it and see what errors you run into? Right off the bat, the Workspaces bar will not work, this is Sway specific.

**Reply by VarLad** [reply]
@ldelossa I'm getting: 
```
** (way-shell:29930): ERROR **: 11:51:37.893: window_manager_service_sway.c:wm_service_sway_init failed to find socket path.
SIGTRAP: child process 'way-shell' core dumped
```

**Reply by ldelossa** [reply]
This is not a top priority for me, but if you want, we can walk thru editing the code and doing some testing to see if the base set of functionality can work. 

Are you up for that? 

The output you sent is kinda what I expected, we can't find the Sway socket. Since this project is mainly focusing on Sway at the moment.

**Reply by VarLad** [reply]
@ldelossa I can try. It would be nice if I can provide Niri support out of the box. 
Do you know where to start looking? Never done anything like this. Niri does support IPC by the way

**Reply by ldelossa** [reply]
For full support to occur, I need to spend some time abstracting the window manager services which is basically hard coded to Sway. 

For the time being tho, you'd basically want to make Way-Shell work without needing Sway. So the first thing to do would be remove the hard coded sway wm init:
https://github.com/ldelossa/way-shell/blob/main/src/main.c#L84C1-L87C6

And then, we probably don't want to load the workspaces bar into the panel anymore:
https://github.com/ldelossa/way-shell/blob/a1ad826a159d905bc10e41f6fcd97108cbbe27cd/src/panel/panel.c#L147 

So comment out that line above.

**Reply by ldelossa** [reply]
I'm going to start a branch for abstracting the window manager stuff out, and once this is done, I can let you know. At this point it should be just a matter of interfacing with Niri's IPC and implementing the interface methods. I'll keep you posted on this :)

**Reply by ldelossa** [reply]
https://github.com/ldelossa/way-shell/pull/37

Here is a PR for window manager abstraction.

**Reply by dchinmay2** [reply]
#68
